Libraries/Microsoft.Teams.Apps/Plugins/Plugin.cs

```diff
@@ -8,6 +8,8 @@ namespace Microsoft.Teams.Apps.Plugins;
 /// </summary>
 public interface IPlugin
 {
+    public IDictionary<string, object>? ContextExtra { get; set; }
```

Comment:
should this be a `buildContextExtra`? since it might be different for each execution?

Reply:
hmm if its per request, maybe we can somehow denote that?

for the `extra` part`, it might be a little trippy where extra denotes "additional" but it can also override old entries

Reply:
left a comment but this should be on a request basis and plugins are initialized on startup, we don't want things to be global we want to add a way to set this extra key/value data on a per-request basis via the existing `Do` sender method https://github.com/microsoft/teams.net/blob/main/Libraries/Microsoft.Teams.Apps/Plugins/SenderPlugin.cs#L45 (which should be an overload since some may not want to set that extra data)